### PR TITLE
Add support for Podcast RSS feed imports

### PR DESCRIPTION
**Pull request by nicktmro**
This is very similar to the existing RSS importer. Unlike the RSS importer, this importer will also extract the `enclosure` tag and wrap the url into an `audio` HTML 5 tag so that compliant browsers can just play the audio file.

Standard "shownotes" can be extracted from a feed by specifying the `body` launch argument as per the included documentation.

There is also support to not overwrite previous files using the `overwrite` launch argument with a `false` value. The default is to overwrite any existing files.

### CODE REVIEW

**Comment by nicktmro** [comment]
Done

**Comment by nicktmro** [comment]
@ashmaroli I tried extending the RSS Importer (as per your instructions) and leaving just the `def self.process(options)`, since it's where all my changes went, but I get this:
```
rss_podcast.rb:5:in `<module:Importers>': uninitialized constant JekyllImport::Importers::RSS (NameError)
```

As I mentioned in #412 I am not a Ruby dev. My attempts to import / reference the RSS.rb file have all failed, and when I checked all the other Importers, I noticed that none of them actually extend anything other than `Importer` so I coudn't follow a pattern.  My googling is also taking me down rabbit holes and I was wondering if you could shed more light on how to solve the error above?

**Comment by ashmaroli** [comment]
@nicktmro Perhaps the error you're seeing is because the `rss_podcast.rb` is being run before `rss.rb` has been loaded into memory.
So lets try an ensure that the `rss.rb` has been loaded beforehand by adding a call to `require_relative` at the top of the `rss_podcast.rb` file:
```ruby
# frozen_string_literal: true

require_relative "rss"

module JekyllImport
  module Importers
    class RSSPodcast < RSS
      # new definitions
    end
  end
end
```
If the above works, do make the change.
Otherwise, let me know and I shall remove the request to make changes to this PR.
Thank you.

**Comment by nicktmro** [comment]
That did the trick :)

File: lib/jekyll-import/importers/rss_podcast.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      # Returns nothing.
+      def self.process(options)
+        source = options.fetch("source")
+        frontmatter = options.fetch("frontmatter", [])
+        body = options.fetch("body", ["description"])
+        overwrite = options.fetch("overwrite", true)
+
+        content = ""
+        uri = URI.parse(source)
+        uri.open { |s| content = s.read }
```
Comment (ashmaroli) [inline lib/jekyll-import/importers/rss_podcast.rb]: Any reason you opted for `URI.parse(source).open` instead of `open(source)` used in `rss.rb`?

**Comment by nicktmro** [comment]
CI failed when I used the “open” call.
Something to do with Kernel#open being insecure.

File: lib/jekyll-import/importers/rss_podcast.rb (hunk 1 above)
Comment (nicktmro) [reply lib/jekyll-import/importers/rss_podcast.rb]: CI failed when I used the “open” call.
Something to do with Kernel#open being insecure.

**Comment by ashmaroli** [comment]
The sole changes in this importer as compared to the RSS Importer is actually:
```diff
@@ -30,10 +30,11 @@ module JekyllImport
       # Returns nothing.
       def self.process(options)
         source = options.fetch("source")
         frontmatter = options.fetch("frontmatter", [])
         body = options.fetch("body", ["description"])
+        overwrite = options.fetch("overwrite", true)
 
         content = ""
         open(source) { |s| content = s.read }
         rss = ::RSS::Parser.parse(content, false)
 
@@ -42,10 +43,15 @@ module JekyllImport
         rss.items.each do |item|
           formatted_date = item.date.strftime("%Y-%m-%d")
           post_name = Jekyll::Utils.slugify(item.title, :mode => "latin")
           name = "#{formatted_date}-#{post_name}"
 
+          # Skip this file if it already exists and overwrite is turned off
+          next if !overwrite && File.file?("_posts/#{name}.html")
+
+          audio = item.enclosure.url
+
           header = {
             "layout" => "post",
             "title"  => item.title,
           }
 
@@ -67,10 +73,11 @@ module JekyllImport
           FileUtils.mkdir_p("_posts")
 
           File.open("_posts/#{name}.html", "w") do |f|
             f.puts header.to_yaml
             f.puts "---\n\n"
+            f.puts "<audio controls=\"\"><source src=\"#{audio}\" type=\"audio/mpeg\">Your browser does not support the audio element.</audio>"
             f.puts output
           end
         end
       end
     end
```
The rest are identical.. Perhaps we should just "enhance the existing RSS importer" with some `options` instead of extending it with a few minor changes.. :thinking:

**Comment by nicktmro** [comment]
That sounds good to me, but I probably lack the skill to make that happen. Happy to help test it or justify the current “options”, though.

**Comment by ashmaroli** [comment]
@nicktmro I've merged your rss_podcast importer into the existing rss importer with a few changes. (Notably, I removed the `overwrite` configuration since I felt that to be outside the scope of this PR's primary intention.)
However, feel free to make changes to suit your original use-case and yeah, update the documentation once the implementation is stable.

**Review by parkr — APPROVED** [review]
:heart:

**Comment by parkr** [comment]
@jekyllbot: merge +minor

**Pull request merged**